src/libcore/cmp.rs

```diff
@@ -625,6 +627,7 @@ pub trait PartialOrd<Rhs: ?Sized = Self>: PartialEq<Rhs> {
     /// let result = std::f64::NAN.partial_cmp(&1.0);
     /// assert_eq!(result, None);
     /// ```
+    #[must_use]
```

Comment:
Intentional; it felt like "part of the set", even if it's not really an operator.